internal/support/scheduler/schedulercontext_test.go

```diff
@@ -182,26 +182,62 @@ func TestIsComplete(t *testing.T) {
 	}
 }
 
-func TestParseFrequency(t *testing.T) {
-	durationStr := "P1D"
-	duration := parseFrequency(durationStr)
+func TestParseNanoSecondFrequency(t *testing.T) {
+
+	durationStr := "50ns"
+	duration, err := parseFrequency(durationStr)
+	if err != nil {
+		t.Errorf(TestUnexpectedMsgFormatStrForInt64Val, duration.Nanoseconds(), 50)
+	}
+	if duration.Nanoseconds() != int64(50) {
+		t.Errorf(TestUnexpectedMsgFormatStrForInt64Val, duration.Nanoseconds(), 50)
+	}
+}
+
+// Note Time.Duration does not support milliseconds, or microseconds directly.
+func TestParseMicrosecondsFrequency(t *testing.T) {
```

Comment:
- Making a recommendation for future unit-testing organization. While these tests are thorough, they could benefit from conciseness by the use of a `table-driven` format (it really helps cuts through repetitive testing steps), and an example of which would be the unit tests in [operation_test.go]( https://github.com/edgexfoundry/edgex-go/blob/master/internal/system/agent/executor/operation_test.go)
- The PR as it stands LGTM 👍 
- Approving the PR, pending building checks following the `rebase`.